tests/ut/backends/test_valkey.py

```diff
@@ -240,3 +240,41 @@ def test_build_key_double_dot(
 
     def test_build_key_no_namespace(self, valkey_cache):
         assert valkey_cache.build_key(Keys.KEY, namespace=None) == Keys.KEY
+
+    def test_custom_serializer(self, valkey_config):
+        assert isinstance(
+            ValkeyCache(config=valkey_config, serializer=PickleSerializer()).serializer,
+            PickleSerializer,
+        )
```

Comment:
Ideally, it'd also be nice to actually exercise the serializer too, so we see it works all the way through (i.e. verify we can save and retrieve something).